kernel/src/process_standard.rs

```diff
@@ -1705,6 +1702,12 @@ impl<C: 'static + Chip> ProcessStandard<'_, C> {
             timeslice_expiration_count: 0,
         });
 
+        if let Some(storage_perm_policy) = storage_permissions_policy {
+            process.storage_permissions = storage_perm_policy.get_permissions(process);
+        } else {
+            process.storage_permissions = StoragePermissions::new_null();
+        }
+
```

Comment:
I don't know whether I like this pattern where we're assigning storage permissions in the `Process` "constructor", and passing the partially constructed `process` reference into this method here. I'm wondering whether it's not better to assign the `storage_permissions` field after process construction, and calling this method in the loader instead?

In my opinion, given the storage permissions policy is generic and receives a `&ProcessStandard` reference, it should be able to expect that reference to be fully constructed.

Reply:
The problem is that ProcessStandard::create() has this signature:

```rust
    pub(crate) unsafe fn create<'a>(
        kernel: &'static Kernel,
        chip: &'static C,
        pb: ProcessBinary,
        remaining_memory: &'a mut [u8],
        fault_policy: &'static dyn ProcessFaultPolicy,
        storage_permissions_policy: Option<&'static dyn ProcessStandardStoragePermissionsPolicy<C>>,
        app_id: ShortId,
        index: usize,
    ) -> Result<(Option<&'static dyn Process>, &'a mut [u8]), (ProcessLoadError, &'a mut [u8])>
```

It returns `&Process` so the caller cannot set `.storage_permissions`.

Reply:
I'm not in _love_ with how this is structured in general, but this approach does balance tock use cases with rust limitations in a perhaps reasonable way.

Reply:
Yes, agreed. I really don't like how the process constructor works here, but this is sound reasoning.

Tangentially, do you know why we're not returning an owned ProcessStandard here that we then move into a static container and take a `&'static dyn Process` reference to? As stated above, it's kind of weird for a constructor to return a reference.

Reply:
Rust is hard and we're just doing our best to make the kernel compile.